### PR TITLE
MS-49: Sessions - API Rest

### PR DESCRIPTION
**Pull request by luisnannini**
Fix package-lock.json
Add date validation
Merge and fix conflicts

### CODE REVIEW

**Comment by luisnannini** [comment]
I didnt touch "package-lock.json" :( how can I fix that..

File: src/controllers/sessions.js [hunk 1]
```diff
@@ -1,84 +1,99 @@
...
-    req.query.dateTime ||
-    !req.query.status ||
-    !req.query.result
-  ) {
-    return res.status(400).send("Some parameters are missing");
-  }
-  return res.status(201).json(sessions);
+
+  Sessions.create(newSession);
+  res.status(201).json(newSession);
```
Comment (nicolasblasco) [inline src/controllers/sessions.js:14]: @luisnannini we should use a callback or a proomise to also send an error response. We could use the same promise we use in getAll.

File: src/controllers/sessions.js [hunk 2]
```diff
@@ -1,84 +1,99 @@
...
 };
 
 const getAll = (req, res) => {
-  return res.status(200).json(sessions);
+  Sessions.find()
+    .then((sessions) => {
+      return res.status(200).json(sessions);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/sessions.js:57]: @luisnannini we should change this HTTP response status code to 404: Not Found. We could do the same for getById, getByIdPsychologist and getByIdCandidate.

File: src/controllers/sessions.js [hunk 3]
```diff
@@ -1,84 +1,99 @@
...
 const getById = (req, res) => {
-  const session = sessions.find((session) => session.id === req.params.id);
-  if (session) return res.status(200).json(session);
-  return res.status(404).json({ msg: `No session with id: ${req.params.id}` });
+  Sessions.findById({ _id: new ObjectId(req.params.id) })
+    .then((session) => {
+      return res.status(200).json(session);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/sessions.js:67]: @luisnannini we should change this HTTP response status code to 404: Not Found.

File: src/controllers/sessions.js [hunk 4]
```diff
@@ -1,84 +1,99 @@
...
-  if (session) return res.status(200).json(session);
-  return res
-    .status(404)
-    .json({ msg: `No psychologist with id: ${req.params.idPsychologist}` });
+  Sessions.findById({ _id: new ObjectId(req.params.id) })
+    .then((session) => {
+      return res.status(200).json(session);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/sessions.js:77]: @luisnannini we should change this HTTP response status code to 404: Not Found.

File: src/controllers/sessions.js (hunk 2 above)
Comment (nicolasblasco) [inline src/controllers/sessions.js:57]: @luisnannini we should change this HTTP response status code to 404: Not Found.

File: src/controllers/sessions.js (hunk 3 above)
Comment (nicolasblasco) [inline src/controllers/sessions.js:67]: @luisnannini we should change this HTTP response status code to 404: Not Found.

File: src/controllers/sessions.js (hunk 4 above)
Comment (nicolasblasco) [inline src/controllers/sessions.js:77]: @luisnannini we should change this HTTP response status code to 404: Not Found.

File: src/controllers/sessions.js [hunk 5]
```diff
@@ -1,84 +1,99 @@
...
+    });
+};
+
+const getByIdCandidate = (req, res) => {
+  Sessions.findById({ _id: new ObjectId(req.params.id) })
+    .then((session) => {
+      return res.status(200).json(session);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/sessions.js:87]: @luisnannini we should change this HTTP response status code to 404: Not Found.

**Review by JamirMansilla — APPROVED** [review]
🚗

File: src/validations/sessions.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  ) {
+    return res.status(400).json({ Msg: "Parameter/s missing" });
+  }
+  next();
+};
+
+const validate = (req, res, next) => {
+  const idPsychologist = req.body.idPsychologist;
+  const idCandidate = req.body.idCandidate;
+  //const dateTime = req.body.dateTime;
```
Comment (damianalvarez) [inline src/validations/sessions.js:16]: @luisnannini I think that to make this work we'll need to use 

```
const dateTime = new Date(req.body.dateTime);
```

**Review by damianalvarez — CHANGES_REQUESTED** [review]

**Comment by luisnannini** [comment]
Fix package-lock.json
Add date validation
Merge and fix conflicts

**Review by damianalvarez — APPROVED** [review]

**Pull request merged**